File: backend/src/main/java/com/pennypulse/backend/domain/RecurringExpense.java
package com.pennypulse.backend.domain;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

@Entity
@Table(name = "recurring_expenses")
public class RecurringExpense {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private Double amount;

    @NotBlank
    private String category;

    private String note;

    @NotNull
    private Integer dayOfMonth;

    private boolean active = true;

    public Long getId() { return id; }
    public Double getAmount() { return amount; }
    public void setAmount(Double amount) { this.amount = amount; }
    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }
    public String getNote() { return note; }
    public void setNote(String note) { this.note = note; }
    public Integer getDayOfMonth() { return dayOfMonth; }
    public void setDayOfMonth(Integer dayOfMonth) { this.dayOfMonth = dayOfMonth; }
    public boolean isActive() { return active; }
    public void setActive(boolean active) { this.active = active; }
}

File: app/src/main/java/com/example/expensetracker/data/RecurringExpense.kt
package com.example.expensetracker.data

data class RecurringExpense(
    val id: Long,
    val amount: Double,
    val category: String,
    val note: String,
    val dayOfMonth: Int,
    val isActive: Boolean
)

fun RecurringExpenseEntity.toDomain(): RecurringExpense {
    return RecurringExpense(
        id = id,
        amount = amount,
        category = category,
        note = note,
        dayOfMonth = dayOfMonth,
        isActive = isActive
    )
}

fun RecurringExpense.toEntity(lastAppliedMonthEpochDay: Long = 0): RecurringExpenseEntity {
    return RecurringExpenseEntity(
        id = id,
        amount = amount,
        category = category,
        note = note,
        dayOfMonth = dayOfMonth,
        isActive = isActive,
        lastAppliedMonthEpochDay = lastAppliedMonthEpochDay
    )
}
